src/private/api/controllers/users.controller.js

```diff
@@ -0,0 +1,426 @@
+    const {
+        username
+    } = req.body;
+
+    const {
+        sql,
+        values
+    } = SQL.USERS.UPDATE('username', id, username);
+    conn.query(sql, values, function (err, result) {
+        if (err) {
+            const duplicateUsername = err.code === ERROR.SQL_DUP_CODE && err.sqlMessage.includes("username");
+            if (duplicateUsername) return respondToClient(res, ERROR.DUPLICATE_USERNAME());
+            return respondToClient(res, err);
+        }
+        if (result.affectedRows === 0) return respondToClient(res, ERROR.INVALID_ENTITY_ID(ENTITY.USER, id));
+
+        respondToClient(res, err, 200);
+    });
+};
+
+/** Change user's password in database */
+exports.changePassword = (req, res) => {
+    const {
+        id
+    } = req.params;
+    const {
+        oldPassword,
+        newPassword
+    } = req.body;
+
+    async.waterfall([
+        function (callback) { // Get current password of user
+            conn.query(SQL.USERS.READ.SINGLE(), id, function (err, [user] = []) {
+                if (err) return callback(err);
+                if (!user) return callback(ERROR.INVALID_ENTITY_ID(ENTITY.USER, id));
+                callback(null, user.password);
+            });
+        },
+        function (password, callback) { // Verify that current password is valid
+            if (!(bcrypt.compareSync(oldPassword, password) || oldPassword === password)) {
+                callback(ERROR.INCORRECT_PASSWORD());
+            } else {
+                callback(null);
+            }
+        },
+        function (callback) { // Hash new password
+            bcrypt.hash(newPassword, 8, function (err, hash) {
+                err ? callback(err) : callback(null, hash);
+            });
+        },
+        function (hash, callback) { // Store new hashed password
+            const {
+                sql,
+                values
+            } = SQL.USERS.UPDATE('password', id, hash);
+            conn.query(sql, values, function (err) {
+                err ? callback(err) : callback(null);
+            });
+        }
+    ], function (err) {
+        respondToClient(res, err, 200);
+    });
+};
+
+/** Change user's clearance */
+exports.changeClearance = (req, res) => {
+    const {
+        id,
+        value
+    } = req.params;
+    const {
+        sql,
+        values
+    } = SQL.USERS.UPDATE('clearance', id, value);
+    conn.query(sql, values, function (err) {
+        respondToClient(res, err, 200);
+    });
+};
+
+/** Delete a user */
+exports.deleteUser = (req, res) => {
+    // TODO: Differentiate between self-deletion and admin-deletion.
+    const id = req.params.id;
+    conn.query(SQL.USERS.DELETE, id, function (err, result) {
+        if (err) return respondToClient(res, err);
+        if (result.affectedRows === 0) err = ERROR.INVALID_ENTITY_ID(ENTITY.USER, id);
+        respondToClient(res, err, 204);
+    });
+};
+
+/** Retrieve all users */
+exports.purgeAddUsers = (req, res) => {
```

Comment:
This function clears (purges) the users records in the User database table.
Change the comment and rename the function to `exports.purgeUsers`.